libtenzir/builtins/functions/community_id.cpp

```diff
@@ -138,8 +138,8 @@ class plugin final : public function_plugin {
       if (src_port_series) {
         src_ports = src_port_series->as<int64_type>();
```

Comment:
Related: The ports should be allowed to be any number type. This function should handle that transparently and call `narrow` on each individual value.

Reply:
Patching this in manually for ports and seed is out of scope for this PR. I think what's clear is that we need some utilities that do this for the developer.

Reply:
As discussed before[^1], you need a slightly different function for that, because you have to communicate that the cast failed and at the same time return the remaining values that could be cast successfully. So `as` cannot do this _as_-is (heh). But yea, we need something like that soon.

[^1]: https://github.com/tenzir/tenzir/pull/4388#issuecomment-2229109987